assets/js/base/components/text-input/style.scss

```
@@ -1,7 +1,23 @@
 .wc-block-text-input {
 	position: relative;
-	margin-top: $gap;
+	margin-bottom: $gap-large;
 	white-space: nowrap;
+
+	&.has-error input {
```

Just an idea, but what do you think about increasing the specificity of this selector like this:
```suggestion
	&.has-error input[type] {
```
This way we can remove the `!important` below. We would still need a comment explaining `[type]` is there for specificity, but that way we avoid an `!important` rule.

---

Ya that makes sense, I'll leave for #1939